### PR TITLE
KIALI-2456,2459,2466,2465, 2450 and 2507 . Improve toolbar for Jaeger Integration and refactor

### PR DESCRIPTION
**Pull request by aljesusg**
- [KIALI-2456](https://issues.jboss.org/browse/KIALI-2456) Error Traces shows kiali home when refreshing the page
- [KIALI-2459](https://issues.jboss.org/browse/KIALI-2459) Fix traces tab if value is -1
- [KIALI-2466](https://issues.jboss.org/browse/KIALI-2466) Remove visual options in jaeger toolbar (Hide graph, hide summary) and make the search button bigger
- [KIALI-2465](https://issues.jboss.org/browse/KIALI-2465) Get Services in jaegerToolbar is not working after Openshift oauth changes
- [KIALI-2450](https://issues.jboss.org/browse/KIALI-2450) Change the namespace Selector in Jaeger Distributed tracing view to be like other views
- [KIALI-2507](https://issues.jboss.org/browse/KIALI-2507) Distributed tracing link disappear in a new tab
- [KIALI-2515](https://issues.jboss.org/browse/KIALI-2515) UI About - Components not persist when the user open a new tab

- All removed from redux to improve performance
- Now when you click lookback custom by default is set the dates from 1 hour
- Now we support query params in the jaeger integration
Screenshot from 2019-03-11 09-34-40
Peek 2019-03-11 09-39

### CODE REVIEW

File: src/store/ConfigStore.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ const namespacePersistFilter = whitelistInputWithInitialState(
 const persistConfig = {
   key: persistKey,
   storage: storage,
-  whitelist: ['authentication', 'namespaces', 'serverConfig'],
+  whitelist: ['authentication', 'namespaces', 'serverConfig', 'jaegerState'],
```
Comment (josejulio) [inline src/store/ConfigStore.ts]: Should we store all the ´jaegerState´? I'm worried about storing information that might not be uptated the next time we use Kiali such as: toolbar.isFetchingService and jaegerURL.

Please, if needed, consider to only store a subset of this information, see: https://github.com/kiali/kiali-ui/blob/master/src/store/ConfigStore.ts#L30-L41

Comment (aljesusg) [reply src/store/ConfigStore.ts]: Totally agree , I am still testing why the state of redux take time in get updated before the rendering of the component, We'll only persist the jaegerUrl.

**Comment by aljesusg** [comment]
@hhovsepy I fixed the problem here, now traces tab is hide until load. But I see another problems like the missing sidecar icon (See the gif)

persiststore

Maybe we should create some loading view while we fetch the backend what do you think @lucasponce @jotak @xeviknal ?

**Comment by aljesusg** [comment]
Removed visual options (Enable by default) to avoid the user get confuse and make the search button more visual. What do you think @pilhuhn ? I think that this is clearer now
screenshot from 2019-02-27 09-48-18

**Comment by pilhuhn** [comment]
@aljesusg Much better.   I personally (but I don't speak for UxD) would even put the search button next to the input boxes and not "far away" in the corner.

**Comment by aljesusg** [comment]
Now we have the same namespace selector
screenshot from 2019-02-28 13-35-37

# Behaviour

- If you have 2 namespaces selected when you go to JaegerTracing view this namespaces are empty.
- If you have 1 namespace selected when you go to JaegerTracing view you will see this namespace selected

File: src/store/ConfigStore.ts [hunk 2]
```diff
@@ -40,11 +40,16 @@ const namespacePersistFilter = whitelistInputWithInitialState(
...
+  createFilter(`jaegerState.search.namespaceSelected`)
+];
+
 const persistConfig = {
   key: persistKey,
   storage: storage,
   whitelist: ['authentication', 'namespaces', 'serverConfig'],
-  transforms: [namespacePersistFilter]
+  transforms: [namespacePersistFilter, ...persitingJaegerReducers]
 };
```
Comment (aljesusg) [inline src/store/ConfigStore.ts]: Now I persist only the data needed @josejulio

**Comment by hhovsepy** [comment]
Tested all the issues mentioned in the description.
Only problem is when kiali is in https and jeager-query  is with http, the Traces tab is hidden.
screenshot from 2019-02-28 13-38-58

**Comment by aljesusg** [comment]
Fixed !!! Thanks !!!

**Comment by aljesusg** [comment]
I keep the same namespace selector, and now you can see all services of the namespace selected like in jaeger view.
peek 2019-03-01 11-28

**Comment by jotak** [comment]
Waw that's a title! :-) Maybe you could set a more meaningful title to better identify this PR in our notifications?

**Comment by aljesusg** [comment]
> Waw that's a title! :-) Maybe you could set a more meaningful title to better identify this PR in our notifications?

So Jiras in the same PR ^^ I'll change to another title with more sense, In the description you can see the JIRAs related. Sorry @jotak ^^

**Comment by hhovsepy** [comment]
Verified all issues mentioned:
Toolbar, services, search button, namespaces, and redirecting when http.

screenshot from 2019-03-01 13-24-31
screenshot from 2019-03-01 13-26-15
screenshot from 2019-03-01 13-29-37

**Comment by lucasponce** [comment]
One thing I see, that can be addressed in a different PR, is that traces tab is not linkable.
Parameters are not added into the url.

**Comment by lucasponce** [comment]
Sometimes the search button doesn't honor the parameters:
image

Same query:
image

**Comment by aljesusg** [comment]
Pr rdy for review @lucasponce @xeviknal @hhovsepy

**Comment by hhovsepy** [comment]
Login page is not opened, it shows empty page, on both browsers.
Screenshot from 2019-03-11 12-48-19

**Comment by aljesusg** [comment]
> Login page is not opened, it shows empty page, on both browsers.
> Screenshot from 2019-03-11 12-48-19

!!! What ??? No console output ? I didn't change nothing from login ^^ I am building the UI to check this

**Comment by aljesusg** [comment]
Ok I found the problem, I had persistStore and I didn't see that. Thanks  @hhovsepy

**Comment by aljesusg** [comment]
Fixed @hhovsepy and I did the JIRA https://issues.jboss.org/browse/KIALI-2515 too, it was related with the status too.

Thanks !!!

**Comment by hhovsepy** [comment]
Tested all the issues mentioned in the description.
Only problem is when kiali is in https and jeager-query is with http, the Traces tab is not redirecting.

**Comment by hhovsepy** [comment]
Screenshot from 2019-03-11 14-05-41

**Comment by hhovsepy** [comment]
All verified now.

**Comment by lucasponce** [comment]
Not a blocker for this PR, but I think we need to document on this Sprint all the steps/troubleshooting related with certificates, like this case:

image

Known issues, or something that is not really a Kiali issue but we can point users when they ask for it.

**Pull request merged**